src/load/migrate-database.lisp

```diff
@@ -255,6 +255,29 @@
     (setf (catalog-distribution-rules catalog)
           (citus-distribute-schema catalog distribute))))
 
+(defun optimize-table-copy-ordering (catalog)
+  "Return a list of tables to copy over in optimized order"
+  (let ((table-list (copy-list (table-list catalog)))
+        (view-list  (copy-list (view-list catalog))))
+    ;; when materialized views are not supported, view-list is empty here
+    (cond
+      ((notevery #'null (mapcar #'table-row-count-estimate table-list))
+           (flet ((compare-row-count-estimates (a b)
+                    (declare (type (or null fixnum) a b))
+                    (> (or a 0) (or b 0))))
+             (let ((sorted-table-list (sort table-list
+                                            #'compare-row-count-estimates
+                                            :key #'table-row-count-estimate)))
+               (log-message :notice
+                            "Processing tables in this order: ~{~a: ~d rows~^, ~}"
+                            (loop :for table
+                               :in (append sorted-table-list view-list)
```

Comment:
`sorted-table-list` is fresh and straight out of `sort`; we may use `nconc` instead of `append` in this line.

Reply:
Done.

Note that pgloader makes almost no efficiency efforts when dealing with the catalogs, because that's not where the program spend its time on when running anyway. It's all about moving the data. [Amdahl's law](https://en.wikipedia.org/wiki/Amdahl%27s_law) would tell us that the best we can optimise for in that area of the code might be in the are of 1% to 5% of the run-time, I would guess.